### PR TITLE
Convert material design png icons to vector assets

### PR DESCRIPTION
**Pull request by JordanCleeve**
This is to replace the existing Google material design png icons with their svg equivalents to improve app size and screen scalability.
This has replaced most of the icons used in the app.

This work has been done following the conversion in pull request [#5119](https://github.com/ankidroid/Anki-Android/pull/5119).

### CODE REVIEW

**Comment by mikehardy** [comment]
@JordanCleeve - cool - I'm testing these now, and I see a crash on the API15-API19 emulators but it's not because the idea here isn't good, it's because our FloatingActionButton library isn't pulling drawables the correct way, and some of our code is also doing it wrong

The solution for the FAB is either

1. dump it since it isn't supported (#4936), 
1. fork the library and update it with [the change here](https://stackoverflow.com/a/37253670/9910298)
1. convert everything but the FAB icons

I think leaving the FAB icons alone is probably the best course, and whenever #4936 happens the icons will be converted at the same time so they won't be remnants forever

Additionally, we don't do merge commits normally, the way we develop on AnkiDroid is with rebasing mostly. To fix this branch for testing I did these things in order to have one clean commit with just your changes:

1. git co master
1. git fetch --all
1. git rebase master  (those last three commands get your branch all up to date)
1. git push (that gets github.com up to date as well, for the future)
1. git co convert-icons-to-vector-assets (and make sure it's all clean)
1. git reset --hard HEAD~1  (this peels the last commit off the branch)
1. git rebase master (this pulls all those master changes in to your branch cleanly with no merge commit)
1. git push origin +convert-icons-to-vector-assets  (that clobbers github state with your new rebased instead of merged local branch)

At that point you should have a fully up to date branch, with one clean commit on it, ready for us to merge - and without that, we won't be able to merge it because the commit history will include the merge commit of things already on master

The usual development process is just repeating the master work (fetch / rebase / push) then switching to your dev branch and doing the git rebase master to stay up to date

For FAB on API<=19 think it's just ic_folder_white, ic_add_white, and ic_file_download_white that we must alter

Unfortunately, after I resurrected those files I still had problems - there is something in our code that is not referencing the files through the srcCompat pathway: https://developer.android.com/guide/topics/graphics/vector-drawable-resources#vector-drawables-backward-solution so it's crashing like so:

```
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/ic_chevron_right_black.xml from drawable resource ID #0x7f080087
        at android.content.res.Resources.loadDrawable(Resources.java:1928)
        at android.content.res.TypedArray.getDrawable(TypedArray.java:601)
        at com.ichi2.anki.widgets.DeckAdapter.<init>(DeckAdapter.java:119)
        at com.ichi2.anki.DeckPicker.onCreate(DeckPicker.java:407)
        at android.app.Activity.performCreate(Activity.java:4466)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1920)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1981) 
        at android.app.ActivityThread.access$600(ActivityThread.java:123) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1147) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4424) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #1: invalid drawable tag vector
        at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:863)

```

I'm out of time to follow the trail further but build an API19 emulator to test these changes and you'll see it. We need to make sure all the drawables are loaded through the compat library pathway in order to do vector graphics on API19 and below

**Comment by mikehardy** [comment]
Side note: looks to save about 200Kb between master and my version after resurrecting the FAB PNGs. Not huge but it is something, more importantly the vector graphics look way better

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
+#Sun Mar 10 21:05:59 NZDT 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-all.zip
```
Comment (mikehardy) [inline gradle/wrapper/gradle-wrapper.properties]: I think this is unrelated and even though AndroidStudio pushes it, it increases first-time download by a great deal for new developers

**Comment by mikehardy** [comment]
Hey Hey! I just had a chance to review the new changes (I was traveling before) and it looks like you've found a replacement FAB! That's great because the existing one was unmaintained and this would now also solve #4936 

The merge commit is still a no-go, have you tried the things I mentioned above? Or would you be okay if I pulled your branch on your repo and patched it up then force-pushed it? You'd have to then do a git fetch origin && git rebase --hard origin/convert-icons-to-vector-assets but that's not so bad

I'd love to see this land though and will work with you to make it happen if you like

**Comment by JordanCleeve** [comment]
Hey man, I'm happy for you too pull the branch and fix the last bits up. I'm really busy at the moment so I can't work on it unfortunately. I have looked at some of your comments but haven't committed anything.

On 10/08/2019 12:58 AM, Mike Hardy <notifications@github.com> wrote:

Hey Hey! I just had a chance to review the new changes (I was traveling before) and it looks like you've found a replacement FAB! That's great because the existing one was unmaintained and this would now also solve #4936<https://github.com/ankidroid/Anki-Android/issues/4936>

The merge commit is still a no-go, have you tried the things I mentioned above? Or would you be okay if I pulled your branch on your repo and patched it up then force-pushed it? You'd have to then do a git fetch origin && git rebase --hard origin/convert-icons-to-vector-assets but that's not so bad

I'd love to see this land though and will work with you to make it happen if you like

**Comment by mikehardy** [comment]
I just looked at this again and saw how close it was to being done - I'm going to get this in 2.10 somehow :-)

**Comment by mikehardy** [comment]
I have this mostly shaped up in a local copy of the branch, hoping to finalize the good work @JordanCleeve did before too long - it was close to begin with, so pinning it open

**Comment by mikehardy** [comment]
The difficult FAB part of this is now done, I think my branch my be unblocked! Hope to get to this soon

**Comment by mikehardy** [comment]
I've re-based and re-pushed this one, it's nearly ready to go (survives a click around and a lintPlayRelease check, but it's been so long since it has seen attention I need to do an audit to see if there are any png icons in use

**Comment by mikehardy** [comment]
Yep, at least these, though I don't think we can get rid of the launcher? Unsure on that one, maybe we can

```
mike@isabela:~/work/AnkiDroid/Anki-Android/AnkiDroid/src/main/res (convert-icons-to-vector-assets) % find . | grep ic_|grep -v xml|cut -d'/' -f3|sort|uniq
ic_action_cancel.png
ic_action_suspend_dropdown.png
ic_dialog_alert.png
ic_eraser_variant_white_24dp.png
ic_flip_to_back_white_24px_dropdown.png
ic_launcher.png
ic_launcher_round.png
ic_lookup_normal.png
ic_lookup_pressed.png
ic_menu_add.png
ic_stat_notify.png

```

**Comment by mikehardy** [comment]
Still need to audit more, maybe more of these may be converted but this is the current list

```
mike@isabela:~/work/AnkiDroid/Anki-Android/AnkiDroid/src (convert-icons-to-vector-assets) % find . | grep 'drawable-' |grep png|cut -f5 -d'/'|sort|uniq|grep -v ic_stat_notify|grep -v logo
ankidroid_txt.png
anki.png
av_pause.png
av_play.png
av_rec.png
av_rec_stop.png
av_stop.png
drawer_shadow.9.png
ic_action_suspend_dropdown.png
ic_eraser_variant_white_24dp.png
ic_flip_to_back_white_24px_dropdown.png
spinner_triangle.png
widget_add_note.png
widget_bg_number.9.png
widget_bg_small_finish.png
widget_bg_small.png
widget_bg_time.9.png

```

**Review by mikehardy — APPROVED** [review]
It's just PNG to SVG conversion now, and up to date

**Comment by mikehardy** [comment]
@david-allison-1 if this works for you I suggest we merge as I'm out of time to track down the rest, and make conversion of the rest a new good-first-issue from that last comment

**Comment by david-allison-1** [comment]
> @david-allison-1 if this works for you I suggest we merge as I'm out of time to track down the rest, and make conversion of the rest a new good-first-issue from that last comment

Half is better than nothing. Let's do it

**Comment by mikehardy** [comment]
@david-allison-1 it's more like 95% :-), there's only a few left at this point.

**Comment by mikehardy** [comment]
@JordanCleeve we finally got this in!!!! Yes! :rocket: that only took forever. Thanks for doing this, so long ago. Shaved 150 killobytes off the APK for everyone and it should look better

**Comment by JordanCleeve** [comment]
> @JordanCleeve we finally got this in!!!! Yes! 🚀 that only took forever. Thanks for doing this, so long ago. Shaved 150 killobytes off the APK for everyone and it should look better

Feels good! Nice work team
